data/transactions/logic/eval.go

```diff
@@ -187,10 +194,14 @@ func Eval(program []byte, params EvalParams) (pass bool, err error) {
 			err = PanicError{x, errstr}
 		}
 	}()
-	if (params.Proto != nil) && (params.Proto.LogicSigVersion == 0) {
+	if (params.Proto == nil) || (params.Proto.LogicSigVersion == 0) {
```

Comment:
we could simplify and now that all tests pass a valid Proto declare that params.Proto == nil is a panic or a 500 internal server error or something. one check for Proto != nil at the top and everything else can assume it's there